src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -0,0 +1,765 @@
+                    PayeeACHAccountExtractStep.class, CUPdpParameterConstants.GENERATED_PAYEE_ACH_ACCOUNT_DOC_NOTE_TEXT));
+            
+            // Route the document and send notifications.
+            paatDocument = (MaintenanceDocument) documentService.routeDocument(paatDocument, KFSConstants.EMPTY_STRING, null);
+            sendPayeeACHAccountAddOrUpdateEmail((PayeeACHAccount) paatDocument.getNewMaintainableObject().getDataObject(), payee,
+                    getPayeeACHAccountAddOrUpdateEmailSubject(true), getUnresolvedPayeeACHAccountAddOrUpdateEmailBody(true));
+            return KFSConstants.EMPTY_STRING;
+        } catch (Exception e) {
+            LOG.info("addACHAccount STE " + e.getStackTrace() + e.toString());
+            return getFailRequestMessage(e);
+        }
+    }
+
+    /**
+     * Checks if the existing ACH account matches the data fed in from the input file, and creates and routes
+     * a new PAAT document to edit the account if necessary.
+     * 
+     * NOTE: The current implementation just logs match/mismatch status without making updates.
+     * This will be modified in the future to perform actual updates.
+     */
+    protected String updateACHAccountIfNecessary(Person payee, PayeeACHAccountExtractDetail achDetail, PayeeACHAccount achAccount) {
+        StringBuilder processingResults = new StringBuilder();
+        processingResults.append("Update functionality NOT implmented yet. Data being checked for possible mismatch. ");
+        if (!StringUtils.equals(achDetail.getBankRoutingNumber(), achAccount.getBankRoutingNumber())
+                || !StringUtils.equals(achDetail.getBankAccountNumber(), achAccount.getBankAccountNumber())) {
+            /*
+             * For a future enhancement, we will modify this method to actually update the Payee ACH Account
+             * via a maintenance document, and possibly use more fields for change comparison.
+             */
+            processingResults.append(" Account information from input file for payee ")
+                             .append(achDetail.getNetID())
+                             .append(" and payee type '")
+                             .append(achAccount.getPayeeIdentifierTypeCode())
+                             .append("' does not match what is in KFS, but will be left as-is.");
+        } else {
+            processingResults.append(" Input file's account information for payee ")
+                             .append(achDetail.getNetID())
+                             .append(" of type '")
+                             .append(achAccount.getPayeeIdentifierTypeCode())
+                             .append("' matches what is already in KFS; no updates will be made for this entry.");
+        }
+        processingResults.append(" Update was not performed.");
+        return processingResults.toString();
+    }
+
+    private String getFailRequestMessage(Exception e) {
+        if (e instanceof ValidationException) {

```

Comment:
The issue with a validate-first approach is that the validation could end up being executed twice (once manually and once for the routing), unless we tried to use a non-validating route-document method (which may not be desirable). Not sure how much that would slow things down.